packages/compiler/lib/service.ts

```diff
@@ -11,11 +11,13 @@ interface ServiceDetails {
 }
 
 const programServiceDetails = new WeakMap<Program | Projector, ServiceDetails>();
-function getServiceDetails(program: Program) {
-  let serviceDetails = programServiceDetails.get(program.currentProjector ?? program);
+function getServiceDetails(program: Program | ProjectedProgram) {
+  // TODO do we need this now?
+  const key = "projector" in program ? program.projector : program;
```

Comment:
Would it work to put the service details on program.globalNamespace using the usual state map instead of in a weak map keyed by program | projected program?

Reply:
Could also just key on the program instead of having that.

Reply:
but the global namespace does sound like a better option so we don't have this global state not tied to a program